kubernetes/linux/main.sh

```diff
@@ -5,6 +5,46 @@ startTime=$(date +%s)
 
 echo "startup script start @ $(date +'%Y-%m-%dT%H:%M:%S')"
 
+startAMACoreAgent() {
+      echo "AMACoreAgent: Starting AMA Core Agent since High Log scale mode is enabled"
+
+      AMACALogFileDir="/var/opt/microsoft/linuxmonagent/amaca/log"
+      AMACALogFilePath="$AMACALogFileDir"/amaca.log
+      AMACAConfigFilePath="/etc/opt/microsoft/azuremonitoragent/amacoreagent"
+      export PA_FLUENT_SOCKET_PORT=13000
+      export PA_DATA_PORT=13000
+      export PA_GIG_BRIDGE_MODE=true
+      export GIG_PA_ENABLE_OPTIMIZATION=true
+      export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1
+      export PA_CONFIG_PORT=12563
+      export CounterDataReportFrequencyInMinutes=60
+
+      {
+         echo "export PA_FLUENT_SOCKET_PORT=$PA_FLUENT_SOCKET_PORT"
+         echo "export PA_DATA_PORT=$PA_DATA_PORT"
+         echo "export PA_GIG_BRIDGE_MODE=$PA_GIG_BRIDGE_MODE"
+         echo "export GIG_PA_ENABLE_OPTIMIZATION=$GIG_PA_ENABLE_OPTIMIZATION"
+         echo "export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=$DOTNET_SYSTEM_GLOBALIZATION_INVARIANT"
+         echo "export PA_CONFIG_PORT=$PA_CONFIG_PORT"
+         echo "export CounterDataReportFrequencyInMinutes=$CounterDataReportFrequencyInMinutes"
+      } >> ~/.bashrc
+
+      source ~/.bashrc
+      /opt/microsoft/azure-mdsd/bin/amacoreagent -c $AMACAConfigFilePath --configport $PA_CONFIG_PORT --amacalog $AMACALogFilePath > /dev/null 2>&1 &
+
+      waitforlisteneronTCPport "$PA_FLUENT_SOCKET_PORT" "$WAITTIME_PORT_13000"
+      waitforlisteneronTCPport "$PA_CONFIG_PORT" "$WAITTIME_PORT_12563"
+      # Extract AMACoreAgent version from log file
+      version=""
+      if [ -d "$AMACALogFileDir" ]; then
```

Comment:
should we instead add this in setup.sh? similar to what we are doing for other packages?

Reply:
> should we instead add this in setup.sh? similar to what we are doing for other packages?

> since  AMACA binary part of the same  azure-mdsd package, and we cant get the version of the AMACA and AMACA binary doesnt support --version command properly so we can only get from logs.